brickflow/engine/task.py

```diff
@@ -509,12 +524,15 @@ class ForEachTask(JobsTasksForEachTask):
     TODO riccamini: Add examples
     """
 
-    @field_validator("inputs", mode="before")
-    @classmethod
-    def validate_inputs(cls, inputs: Any) -> str:
-        if not isinstance(inputs, str):
-            inputs = json.dumps(inputs)
-        return inputs
+    configs: JobsTasksForEachTaskConfigs
+    task: Any
+
+    @model_validator(mode="before")
+    def validate_configs(self) -> "ForEachTask":
+        self["inputs"] = self["configs"].inputs  # type: ignore
+        self["concurrency"] = self["configs"].concurrency  # type: ignore
+
+        return self
```

Comment:
Serialization was my only concern but I see you addressed that with a model validator 💪